main.c

```diff
@@ -600,8 +605,12 @@ static void handle_idle(void *data, struct org_kde_kwin_idle_timeout *timer) {
 		set_idle_hint(true);
 	} else
 #endif
-	if (cmd->idle_cmd) {
-		cmd_exec(cmd->idle_cmd);
+	if (!cmd->cond_cmd || cmd_exec(cmd->cond_cmd) == 0) {
```

Comment:
this should probably discriminate between fork failed (negative return value of cmd_exec) and non-zero exit code (should be positive if I understood the waitpid call correctly in that it only returns the least significant 8 bits of the exit code therefore positive (?))